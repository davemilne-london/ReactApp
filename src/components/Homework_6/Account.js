import React, {useReducer, useState} from 'react';
import './Account.module.css';

const countries = ["Afghanistan","Albania","Algeria","Andorra","Angola","Antigua & Deps","Argentina","Armenia","Australia","Austria","Azerbaijan","Bahamas","Bahrain","Bangladesh","Barbados","Belarus","Belgium","Belize","Benin","Bhutan","Bolivia","Bosnia Herzegovina","Botswana","Brazil","Brunei","Bulgaria","Burkina","Burundi","Cambodia","Cameroon","Canada","Cape Verde","Central African Rep","Chad","Chile","China","Colombia","Comoros","Congo","Congo {Democratic Rep}","Costa Rica","Croatia","Cuba","Cyprus","Czech Republic","Denmark","Djibouti","Dominica","Dominican Republic","East Timor","Ecuador","Egypt","El Salvador","Equatorial Guinea","Eritrea","Estonia","Ethiopia","Fiji","Finland","France","Gabon","Gambia","Georgia","Germany","Ghana","Greece","Grenada","Guatemala","Guinea","Guinea-Bissau","Guyana","Haiti","Honduras","Hungary","Iceland","India","Indonesia","Iran","Iraq","Ireland {Republic}","Israel","Italy","Ivory Coast","Jamaica","Japan","Jordan","Kazakhstan","Kenya","Kiribati","Korea North","Korea South","Kosovo","Kuwait","Kyrgyzstan","Laos","Latvia","Lebanon","Lesotho","Liberia","Libya","Liechtenstein","Lithuania","Luxembourg","Macedonia","Madagascar","Malawi","Malaysia","Maldives","Mali","Malta","Marshall Islands","Mauritania","Mauritius","Mexico","Micronesia","Moldova","Monaco","Mongolia","Montenegro","Morocco","Mozambique","Myanmar, {Burma}","Namibia","Nauru","Nepal","Netherlands","New Zealand","Nicaragua","Niger","Nigeria","Norway","Oman","Pakistan","Palau","Panama","Papua New Guinea","Paraguay","Peru","Philippines","Poland","Portugal","Qatar","Romania","Russian Federation","Rwanda","St Kitts & Nevis","St Lucia","Saint Vincent & the Grenadines","Samoa","San Marino","Sao Tome & Principe","Saudi Arabia","Senegal","Serbia","Seychelles","Sierra Leone","Singapore","Slovakia","Slovenia","Solomon Islands","Somalia","South Africa","South Sudan","Spain","Sri Lanka","Sudan","Suriname","Swaziland","Sweden","Switzerland","Syria","Taiwan","Tajikistan","Tanzania","Thailand","Togo","Tonga","Trinidad & Tobago","Tunisia","Turkey","Turkmenistan","Tuvalu","Uganda","Ukraine","United Arab Emirates","United Kingdom","United States","Uruguay","Uzbekistan","Vanuatu","Vatican City","Venezuela","Vietnam","Yemen","Zambia","Zimbabwe"];


const initialState = {
    firstName:"",
    lastName:"",
    email:"",
    password:"",
    country:"",
    terms:""
}

const reducer = (state, action) => {

    if (action.name === "reset") return initialState

    return {
        ...state,
        [action.name]: action.value
    }
}

export function isFormValid (formState) {

    let draftErrors=[];
    for (const[key, value] of Object.entries(formState)) {
        if (value === initialState[key]) {
            draftErrors = [...draftErrors, key];                
        } 
    };

    return draftErrors;
}

const Account = () => {

    const [formState, setFormState] = useReducer(reducer, initialState);
    const [isFormSubmitting, setIsFormSubmitting] = useState(false);
    const [errors, setErrors] = useState([]);

    const resetFormState = () => {
        setFormState({
            name: 'reset'
        });
    }

    const checkErrors = () => {

        const draftErrors = isFormValid(formState);        
        setErrors(draftErrors);
        return errors.length === 0;
    }

    const handleSubmit = (e) => {
        e.preventDefault();       

        if (checkErrors()) {                
            setIsFormSubmitting(true);
            resetFormState();
            setIsFormSubmitting(false);           
        };        
    }

    const handleChange = (e) => {
        
        const value = e.target.type === "checkbox" ? e.target.checked : e.target.value;
        setFormState({
            name: e.target.name,
            value: value
        });
    }

    const {firstName, lastName, email, password, country, terms} = formState;

    return (
        <form onSubmit={handleSubmit}>
            <fieldset className="fieldset">
                <legend>Account</legend>

                <div>
                    <label htmlFor="firstName" className="label">First name:</label>
                    <input 
                        type="text" 
                        id="firstName" 
                        name="firstName"
                        className="formInput"
                        onChange={handleChange}
                        value={firstName}
                    />
                </div>

                <div>
                    <label htmlFor="lastName" className="label">Last name:</label>
                    <input 
                        type="text" 
                        id="lastName" 
                        name="lastName"
                        className="formInput"
                        onChange={handleChange}    
                        value={lastName}
                    />
                </div>

                <div>
                    <label htmlFor="email" className="label">Email:</label>
                    <input 
                        type="text" 
                        id="email" 
                        name="email"
                        className="formInput"
                        onChange={handleChange}
                        value={email}
                    />
                </div>

                <div>
                    <label htmlFor="password" className="label">Password</label>
                    <input 
                        type="password" 
                        id="password" 
                        name="password"
                        className="formInput"
                        onChange={handleChange}
                        value={password}    
                    />
                </div>

                <div>
                    <label htmlFor="country" className="label">Country</label>
                    <select 
                        id="country" 
                        onChange={handleChange}
                        name="country"
                        value={country}
                    >
                        <option key="0" value="0"></option>
                        {countries.map((x,y) => <option key={y}>{x}</option>)};
                    </select>
                </div>

                <div>
                    <input 
                        type="checkbox" 
                        id="terms" 
                        name="terms"
                        onChange={handleChange}
                        checked={terms}
                    />
                    <label htmlFor="terms">I Accept the terms of service</label>
                </div>

                <input type="submit" value="Submit" disabled={isFormSubmitting} className="submit"/>
            </fieldset>

            {errors.length > 0 && (
                <ul>Errors: 
                    {errors.map((error, key) => {
                        return <li key={key}>{error}</li>
                    })}
                </ul>)
            }
        </form>
    )
}

export default Account;