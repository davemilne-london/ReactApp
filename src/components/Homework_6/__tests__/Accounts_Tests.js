import { render } from "@testing-library/react";
import React from 'react';
import Account, {isFormValid} from '../Account';

describe("The display element",() => {
    it("should not regress",() =>{
        const tree = render(<Account />);

        expect (tree).toMatchSnapshot();
    })
})


describe("The isValid function",() => {

    it("should return false if the first name is not set",() =>{      
        //set up state
        const testInvalidState = {
            firstName:"",
            lastName:"test",
            email:"test",
            password:"test",
            country:"test",
            terms:true
        }

        const errors = isFormValid(testInvalidState);
        expect(errors).toHaveLength(1);
    });

    it("should return true if everything is set",() =>{      
        //set up state
        const testValidState = {
            firstName:"",
            lastName:"test",
            email:"test",
            password:"test",
            country:"test",
            terms:true
        }

        const errors = isFormValid(testValidState);
        expect(errors).toHaveLength(1);
    })

});