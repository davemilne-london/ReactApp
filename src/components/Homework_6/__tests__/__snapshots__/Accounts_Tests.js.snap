// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The display element should not regress 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <fieldset
          class="fieldset"
        >
          <legend>
            Account
          </legend>
          <div>
            <label
              class="label"
              for="firstName"
            >
              First name:
            </label>
            <input
              class="formInput"
              id="firstName"
              name="firstName"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              class="label"
              for="lastName"
            >
              Last name:
            </label>
            <input
              class="formInput"
              id="lastName"
              name="lastName"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              class="label"
              for="email"
            >
              Email:
            </label>
            <input
              class="formInput"
              id="email"
              name="email"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              class="label"
              for="password"
            >
              Password
            </label>
            <input
              class="formInput"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
          <div>
            <label
              class="label"
              for="country"
            >
              Country
            </label>
            <select
              id="country"
              name="country"
            >
              <option
                value="0"
              />
              <option>
                Afghanistan
              </option>
              <option>
                Albania
              </option>
              <option>
                Algeria
              </option>
              <option>
                Andorra
              </option>
              <option>
                Angola
              </option>
              <option>
                Antigua & Deps
              </option>
              <option>
                Argentina
              </option>
              <option>
                Armenia
              </option>
              <option>
                Australia
              </option>
              <option>
                Austria
              </option>
              <option>
                Azerbaijan
              </option>
              <option>
                Bahamas
              </option>
              <option>
                Bahrain
              </option>
              <option>
                Bangladesh
              </option>
              <option>
                Barbados
              </option>
              <option>
                Belarus
              </option>
              <option>
                Belgium
              </option>
              <option>
                Belize
              </option>
              <option>
                Benin
              </option>
              <option>
                Bhutan
              </option>
              <option>
                Bolivia
              </option>
              <option>
                Bosnia Herzegovina
              </option>
              <option>
                Botswana
              </option>
              <option>
                Brazil
              </option>
              <option>
                Brunei
              </option>
              <option>
                Bulgaria
              </option>
              <option>
                Burkina
              </option>
              <option>
                Burundi
              </option>
              <option>
                Cambodia
              </option>
              <option>
                Cameroon
              </option>
              <option>
                Canada
              </option>
              <option>
                Cape Verde
              </option>
              <option>
                Central African Rep
              </option>
              <option>
                Chad
              </option>
              <option>
                Chile
              </option>
              <option>
                China
              </option>
              <option>
                Colombia
              </option>
              <option>
                Comoros
              </option>
              <option>
                Congo
              </option>
              <option>
                Congo {Democratic Rep}
              </option>
              <option>
                Costa Rica
              </option>
              <option>
                Croatia
              </option>
              <option>
                Cuba
              </option>
              <option>
                Cyprus
              </option>
              <option>
                Czech Republic
              </option>
              <option>
                Denmark
              </option>
              <option>
                Djibouti
              </option>
              <option>
                Dominica
              </option>
              <option>
                Dominican Republic
              </option>
              <option>
                East Timor
              </option>
              <option>
                Ecuador
              </option>
              <option>
                Egypt
              </option>
              <option>
                El Salvador
              </option>
              <option>
                Equatorial Guinea
              </option>
              <option>
                Eritrea
              </option>
              <option>
                Estonia
              </option>
              <option>
                Ethiopia
              </option>
              <option>
                Fiji
              </option>
              <option>
                Finland
              </option>
              <option>
                France
              </option>
              <option>
                Gabon
              </option>
              <option>
                Gambia
              </option>
              <option>
                Georgia
              </option>
              <option>
                Germany
              </option>
              <option>
                Ghana
              </option>
              <option>
                Greece
              </option>
              <option>
                Grenada
              </option>
              <option>
                Guatemala
              </option>
              <option>
                Guinea
              </option>
              <option>
                Guinea-Bissau
              </option>
              <option>
                Guyana
              </option>
              <option>
                Haiti
              </option>
              <option>
                Honduras
              </option>
              <option>
                Hungary
              </option>
              <option>
                Iceland
              </option>
              <option>
                India
              </option>
              <option>
                Indonesia
              </option>
              <option>
                Iran
              </option>
              <option>
                Iraq
              </option>
              <option>
                Ireland {Republic}
              </option>
              <option>
                Israel
              </option>
              <option>
                Italy
              </option>
              <option>
                Ivory Coast
              </option>
              <option>
                Jamaica
              </option>
              <option>
                Japan
              </option>
              <option>
                Jordan
              </option>
              <option>
                Kazakhstan
              </option>
              <option>
                Kenya
              </option>
              <option>
                Kiribati
              </option>
              <option>
                Korea North
              </option>
              <option>
                Korea South
              </option>
              <option>
                Kosovo
              </option>
              <option>
                Kuwait
              </option>
              <option>
                Kyrgyzstan
              </option>
              <option>
                Laos
              </option>
              <option>
                Latvia
              </option>
              <option>
                Lebanon
              </option>
              <option>
                Lesotho
              </option>
              <option>
                Liberia
              </option>
              <option>
                Libya
              </option>
              <option>
                Liechtenstein
              </option>
              <option>
                Lithuania
              </option>
              <option>
                Luxembourg
              </option>
              <option>
                Macedonia
              </option>
              <option>
                Madagascar
              </option>
              <option>
                Malawi
              </option>
              <option>
                Malaysia
              </option>
              <option>
                Maldives
              </option>
              <option>
                Mali
              </option>
              <option>
                Malta
              </option>
              <option>
                Marshall Islands
              </option>
              <option>
                Mauritania
              </option>
              <option>
                Mauritius
              </option>
              <option>
                Mexico
              </option>
              <option>
                Micronesia
              </option>
              <option>
                Moldova
              </option>
              <option>
                Monaco
              </option>
              <option>
                Mongolia
              </option>
              <option>
                Montenegro
              </option>
              <option>
                Morocco
              </option>
              <option>
                Mozambique
              </option>
              <option>
                Myanmar, {Burma}
              </option>
              <option>
                Namibia
              </option>
              <option>
                Nauru
              </option>
              <option>
                Nepal
              </option>
              <option>
                Netherlands
              </option>
              <option>
                New Zealand
              </option>
              <option>
                Nicaragua
              </option>
              <option>
                Niger
              </option>
              <option>
                Nigeria
              </option>
              <option>
                Norway
              </option>
              <option>
                Oman
              </option>
              <option>
                Pakistan
              </option>
              <option>
                Palau
              </option>
              <option>
                Panama
              </option>
              <option>
                Papua New Guinea
              </option>
              <option>
                Paraguay
              </option>
              <option>
                Peru
              </option>
              <option>
                Philippines
              </option>
              <option>
                Poland
              </option>
              <option>
                Portugal
              </option>
              <option>
                Qatar
              </option>
              <option>
                Romania
              </option>
              <option>
                Russian Federation
              </option>
              <option>
                Rwanda
              </option>
              <option>
                St Kitts & Nevis
              </option>
              <option>
                St Lucia
              </option>
              <option>
                Saint Vincent & the Grenadines
              </option>
              <option>
                Samoa
              </option>
              <option>
                San Marino
              </option>
              <option>
                Sao Tome & Principe
              </option>
              <option>
                Saudi Arabia
              </option>
              <option>
                Senegal
              </option>
              <option>
                Serbia
              </option>
              <option>
                Seychelles
              </option>
              <option>
                Sierra Leone
              </option>
              <option>
                Singapore
              </option>
              <option>
                Slovakia
              </option>
              <option>
                Slovenia
              </option>
              <option>
                Solomon Islands
              </option>
              <option>
                Somalia
              </option>
              <option>
                South Africa
              </option>
              <option>
                South Sudan
              </option>
              <option>
                Spain
              </option>
              <option>
                Sri Lanka
              </option>
              <option>
                Sudan
              </option>
              <option>
                Suriname
              </option>
              <option>
                Swaziland
              </option>
              <option>
                Sweden
              </option>
              <option>
                Switzerland
              </option>
              <option>
                Syria
              </option>
              <option>
                Taiwan
              </option>
              <option>
                Tajikistan
              </option>
              <option>
                Tanzania
              </option>
              <option>
                Thailand
              </option>
              <option>
                Togo
              </option>
              <option>
                Tonga
              </option>
              <option>
                Trinidad & Tobago
              </option>
              <option>
                Tunisia
              </option>
              <option>
                Turkey
              </option>
              <option>
                Turkmenistan
              </option>
              <option>
                Tuvalu
              </option>
              <option>
                Uganda
              </option>
              <option>
                Ukraine
              </option>
              <option>
                United Arab Emirates
              </option>
              <option>
                United Kingdom
              </option>
              <option>
                United States
              </option>
              <option>
                Uruguay
              </option>
              <option>
                Uzbekistan
              </option>
              <option>
                Vanuatu
              </option>
              <option>
                Vatican City
              </option>
              <option>
                Venezuela
              </option>
              <option>
                Vietnam
              </option>
              <option>
                Yemen
              </option>
              <option>
                Zambia
              </option>
              <option>
                Zimbabwe
              </option>
              ;
            </select>
          </div>
          <div>
            <input
              id="terms"
              name="terms"
              type="checkbox"
            />
            <label
              for="terms"
            >
              I Accept the terms of service
            </label>
          </div>
          <input
            class="submit"
            type="submit"
            value="Submit"
          />
        </fieldset>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <fieldset
        class="fieldset"
      >
        <legend>
          Account
        </legend>
        <div>
          <label
            class="label"
            for="firstName"
          >
            First name:
          </label>
          <input
            class="formInput"
            id="firstName"
            name="firstName"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="label"
            for="lastName"
          >
            Last name:
          </label>
          <input
            class="formInput"
            id="lastName"
            name="lastName"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="label"
            for="email"
          >
            Email:
          </label>
          <input
            class="formInput"
            id="email"
            name="email"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            class="label"
            for="password"
          >
            Password
          </label>
          <input
            class="formInput"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div>
          <label
            class="label"
            for="country"
          >
            Country
          </label>
          <select
            id="country"
            name="country"
          >
            <option
              value="0"
            />
            <option>
              Afghanistan
            </option>
            <option>
              Albania
            </option>
            <option>
              Algeria
            </option>
            <option>
              Andorra
            </option>
            <option>
              Angola
            </option>
            <option>
              Antigua & Deps
            </option>
            <option>
              Argentina
            </option>
            <option>
              Armenia
            </option>
            <option>
              Australia
            </option>
            <option>
              Austria
            </option>
            <option>
              Azerbaijan
            </option>
            <option>
              Bahamas
            </option>
            <option>
              Bahrain
            </option>
            <option>
              Bangladesh
            </option>
            <option>
              Barbados
            </option>
            <option>
              Belarus
            </option>
            <option>
              Belgium
            </option>
            <option>
              Belize
            </option>
            <option>
              Benin
            </option>
            <option>
              Bhutan
            </option>
            <option>
              Bolivia
            </option>
            <option>
              Bosnia Herzegovina
            </option>
            <option>
              Botswana
            </option>
            <option>
              Brazil
            </option>
            <option>
              Brunei
            </option>
            <option>
              Bulgaria
            </option>
            <option>
              Burkina
            </option>
            <option>
              Burundi
            </option>
            <option>
              Cambodia
            </option>
            <option>
              Cameroon
            </option>
            <option>
              Canada
            </option>
            <option>
              Cape Verde
            </option>
            <option>
              Central African Rep
            </option>
            <option>
              Chad
            </option>
            <option>
              Chile
            </option>
            <option>
              China
            </option>
            <option>
              Colombia
            </option>
            <option>
              Comoros
            </option>
            <option>
              Congo
            </option>
            <option>
              Congo {Democratic Rep}
            </option>
            <option>
              Costa Rica
            </option>
            <option>
              Croatia
            </option>
            <option>
              Cuba
            </option>
            <option>
              Cyprus
            </option>
            <option>
              Czech Republic
            </option>
            <option>
              Denmark
            </option>
            <option>
              Djibouti
            </option>
            <option>
              Dominica
            </option>
            <option>
              Dominican Republic
            </option>
            <option>
              East Timor
            </option>
            <option>
              Ecuador
            </option>
            <option>
              Egypt
            </option>
            <option>
              El Salvador
            </option>
            <option>
              Equatorial Guinea
            </option>
            <option>
              Eritrea
            </option>
            <option>
              Estonia
            </option>
            <option>
              Ethiopia
            </option>
            <option>
              Fiji
            </option>
            <option>
              Finland
            </option>
            <option>
              France
            </option>
            <option>
              Gabon
            </option>
            <option>
              Gambia
            </option>
            <option>
              Georgia
            </option>
            <option>
              Germany
            </option>
            <option>
              Ghana
            </option>
            <option>
              Greece
            </option>
            <option>
              Grenada
            </option>
            <option>
              Guatemala
            </option>
            <option>
              Guinea
            </option>
            <option>
              Guinea-Bissau
            </option>
            <option>
              Guyana
            </option>
            <option>
              Haiti
            </option>
            <option>
              Honduras
            </option>
            <option>
              Hungary
            </option>
            <option>
              Iceland
            </option>
            <option>
              India
            </option>
            <option>
              Indonesia
            </option>
            <option>
              Iran
            </option>
            <option>
              Iraq
            </option>
            <option>
              Ireland {Republic}
            </option>
            <option>
              Israel
            </option>
            <option>
              Italy
            </option>
            <option>
              Ivory Coast
            </option>
            <option>
              Jamaica
            </option>
            <option>
              Japan
            </option>
            <option>
              Jordan
            </option>
            <option>
              Kazakhstan
            </option>
            <option>
              Kenya
            </option>
            <option>
              Kiribati
            </option>
            <option>
              Korea North
            </option>
            <option>
              Korea South
            </option>
            <option>
              Kosovo
            </option>
            <option>
              Kuwait
            </option>
            <option>
              Kyrgyzstan
            </option>
            <option>
              Laos
            </option>
            <option>
              Latvia
            </option>
            <option>
              Lebanon
            </option>
            <option>
              Lesotho
            </option>
            <option>
              Liberia
            </option>
            <option>
              Libya
            </option>
            <option>
              Liechtenstein
            </option>
            <option>
              Lithuania
            </option>
            <option>
              Luxembourg
            </option>
            <option>
              Macedonia
            </option>
            <option>
              Madagascar
            </option>
            <option>
              Malawi
            </option>
            <option>
              Malaysia
            </option>
            <option>
              Maldives
            </option>
            <option>
              Mali
            </option>
            <option>
              Malta
            </option>
            <option>
              Marshall Islands
            </option>
            <option>
              Mauritania
            </option>
            <option>
              Mauritius
            </option>
            <option>
              Mexico
            </option>
            <option>
              Micronesia
            </option>
            <option>
              Moldova
            </option>
            <option>
              Monaco
            </option>
            <option>
              Mongolia
            </option>
            <option>
              Montenegro
            </option>
            <option>
              Morocco
            </option>
            <option>
              Mozambique
            </option>
            <option>
              Myanmar, {Burma}
            </option>
            <option>
              Namibia
            </option>
            <option>
              Nauru
            </option>
            <option>
              Nepal
            </option>
            <option>
              Netherlands
            </option>
            <option>
              New Zealand
            </option>
            <option>
              Nicaragua
            </option>
            <option>
              Niger
            </option>
            <option>
              Nigeria
            </option>
            <option>
              Norway
            </option>
            <option>
              Oman
            </option>
            <option>
              Pakistan
            </option>
            <option>
              Palau
            </option>
            <option>
              Panama
            </option>
            <option>
              Papua New Guinea
            </option>
            <option>
              Paraguay
            </option>
            <option>
              Peru
            </option>
            <option>
              Philippines
            </option>
            <option>
              Poland
            </option>
            <option>
              Portugal
            </option>
            <option>
              Qatar
            </option>
            <option>
              Romania
            </option>
            <option>
              Russian Federation
            </option>
            <option>
              Rwanda
            </option>
            <option>
              St Kitts & Nevis
            </option>
            <option>
              St Lucia
            </option>
            <option>
              Saint Vincent & the Grenadines
            </option>
            <option>
              Samoa
            </option>
            <option>
              San Marino
            </option>
            <option>
              Sao Tome & Principe
            </option>
            <option>
              Saudi Arabia
            </option>
            <option>
              Senegal
            </option>
            <option>
              Serbia
            </option>
            <option>
              Seychelles
            </option>
            <option>
              Sierra Leone
            </option>
            <option>
              Singapore
            </option>
            <option>
              Slovakia
            </option>
            <option>
              Slovenia
            </option>
            <option>
              Solomon Islands
            </option>
            <option>
              Somalia
            </option>
            <option>
              South Africa
            </option>
            <option>
              South Sudan
            </option>
            <option>
              Spain
            </option>
            <option>
              Sri Lanka
            </option>
            <option>
              Sudan
            </option>
            <option>
              Suriname
            </option>
            <option>
              Swaziland
            </option>
            <option>
              Sweden
            </option>
            <option>
              Switzerland
            </option>
            <option>
              Syria
            </option>
            <option>
              Taiwan
            </option>
            <option>
              Tajikistan
            </option>
            <option>
              Tanzania
            </option>
            <option>
              Thailand
            </option>
            <option>
              Togo
            </option>
            <option>
              Tonga
            </option>
            <option>
              Trinidad & Tobago
            </option>
            <option>
              Tunisia
            </option>
            <option>
              Turkey
            </option>
            <option>
              Turkmenistan
            </option>
            <option>
              Tuvalu
            </option>
            <option>
              Uganda
            </option>
            <option>
              Ukraine
            </option>
            <option>
              United Arab Emirates
            </option>
            <option>
              United Kingdom
            </option>
            <option>
              United States
            </option>
            <option>
              Uruguay
            </option>
            <option>
              Uzbekistan
            </option>
            <option>
              Vanuatu
            </option>
            <option>
              Vatican City
            </option>
            <option>
              Venezuela
            </option>
            <option>
              Vietnam
            </option>
            <option>
              Yemen
            </option>
            <option>
              Zambia
            </option>
            <option>
              Zimbabwe
            </option>
            ;
          </select>
        </div>
        <div>
          <input
            id="terms"
            name="terms"
            type="checkbox"
          />
          <label
            for="terms"
          >
            I Accept the terms of service
          </label>
        </div>
        <input
          class="submit"
          type="submit"
          value="Submit"
        />
      </fieldset>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
